tests/aws/services/apigateway/test_apigateway_lambda_cfn.py

```diff
@@ -1,80 +1,304 @@
+                            "Key": "connection-header-param",
+                            "Value": "value",
+                            "IsValueSecret": False,
+                        },
+                        {"Key": "overwritten-header", "Value": "original", "IsValueSecret": False},
+                        {"Key": "oauthheader", "Value": "value2", "IsValueSecret": False},
+                    ],
+                    "QueryStringParameters": [
+                        {"Key": "connection_query_param", "Value": "value", "IsValueSecret": False},
+                        {"Key": "overwritten_query", "Value": "original", "IsValueSecret": False},
+                        {"Key": "oauthquery", "Value": "value3", "IsValueSecret": False},
+                    ],
+                },
+            },
+        )
+
+        if auth_type == "OAUTH_CLIENT_CREDENTIALS":
+
+            def _wait_for_connection_auth():
+                conn_desc = aws_client.events.describe_connection(Name=connection_name)
+                state = conn_desc.get("ConnectionState")
+                if state == "AUTHORIZED":
+                    return conn_desc
+                raise AssertionError(f"Connection not yet authorized. Current state: {state}")
+
+            retry(_wait_for_connection_auth, retries=20, sleep=5)
+
+        dest_name = f"d-{short_uid()}"
+        dest_resp = aws_client.events.create_api_destination(
+            Name=dest_name,
+            ConnectionArn=connection_resp["ConnectionArn"],
+            InvocationEndpoint=invocation_endpoint,
+            HttpMethod="POST",
+        )
+        api_destination_arn = dest_resp["ApiDestinationArn"]
+
+        time.sleep(10)
+
+        rule_name = f"r-{short_uid()}"
+        target_id = f"target-{short_uid()}"
+        pattern = json.dumps({"source": ["source-123"], "detail-type": ["type-123"]})
+        aws_client.events.put_rule(Name=rule_name, EventPattern=pattern)
+        time.sleep(10)
+
+        unique_marker = str(uuid.uuid4())
+
+        snapshot.add_transformer(snapshot.transform.regex(unique_marker, "<unique-marker>"))
+
+        detail = {"i": 0, "marker": unique_marker}
+
+        aws_client.events.put_targets(
+            Rule=rule_name,
+            Targets=[
+                {
+                    "Id": target_id,
+                    "Arn": api_destination_arn,
+                    "Input": json.dumps({"target_value": "value", "marker": unique_marker}),
+                    "RoleArn": role_arn,
+                    "HttpParameters": {
+                        "HeaderParameters": {"target-header": "target_header_value"},
+                        "QueryStringParameters": {"target_query": "t_query"},
+                    },
+                }
+            ],
+        )
+
+        try:
+            aws_client.logs.create_log_group(logGroupName=f"/aws/lambda/{backend_function_name}")
+        except aws_client.logs.exceptions.ResourceAlreadyExistsException:
+            pass
```

Comment:
You can use `contextlib.suppress` to achieve a similar goal:

```suggestion
        with contextlib.suppress(aws_client.logs.exceptions.ResourceAlreadyExistsException):
            aws_client.logs.create_log_group(logGroupName=f"/aws/lambda/{backend_function_name}")
```